plasmapy/formulary/ionization.py

```diff
@@ -192,3 +192,93 @@ def Saha(g_j, g_k, n_e: u.m ** -3, E_jk: u.J, T_e: u.K) -> u.dimensionless_unsca
     ratio = degeneracy_factor * physical_constants * boltzmann_factor
 
     return ratio
+
+
+@validate_quantities(
+    n_e={"can_be_negative": False},
+    T_e={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+)
+def thomas_fermi_ionization_state(
+    Z, n_e: u.m ** (-3), T_e: u.K
+) -> u.dimensionless_unscaled:
+    r"""
+    Return the finite temperature Thomas-Fermi mean ionization state using fit provided in
+    R.M. More, "Pressure Ionization, Resonances, and the
+    Continuity of Bound and Free States", Adv. in Atomic
+    Mol. Phys., Vol. 21, p. 332 (Table IV).
+
+    Parameters
+    ----------
+    Z : int
+        Ion charge number
+
+    n_e : `~astropy.units.Quantity`
+        Electron number density in 1/m**3
+
+    T_e : `~astropy.units.Quantity`
+        Electon temperature in K
```

Comment:
```suggestion
        Electron temperature in K or energy per electron.
```
...since `validate_quantities` allows us to input temperature in `eV` per particle too.